src/System.Management.Automation/engine/NativeCommandProcessor.cs

```diff
@@ -545,15 +545,14 @@ private void InitNativeProcess()
             Exception exceptionToRethrow = null;
             try
             {
-                // If this process is being run standalone, tell the host, which might want
-                // to save off the window title or other such state as might be tweaked by
-                // the native process
+                // Before start the executable, tell the host, which might want to save off the
+                // window title or other such state as might be tweaked by the native process.
+                Command.Context.EngineHostInterface.NotifyBeginApplication();
+                _hasNotifiedBeginApplication = true;
+
                 if (_runStandAlone)
                 {
-                    this.Command.Context.EngineHostInterface.NotifyBeginApplication();
-                    _hasNotifiedBeginApplication = true;
```

Comment:
If a native process tweaks the window title, it will do so no matter it's running `StandAlone` or in a pipeline. So, we should actually always call `NotifyBeginApplication` and `NotifyEndApplication` before and after running a native command. Therefore, I moved this code out of the `if (_runStandAlone)` block.